src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/cost/RouteCostCalculator.java

```diff
@@ -0,0 +1,76 @@
+package org.opentripplanner.routing.algorithm.raptoradapter.transit.cost;
+
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.opentripplanner.routing.algorithm.raptoradapter.transit.TripSchedule;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.raptor.api.transit.CostCalculator;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransfer;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransferConstraint;
+
+public class RouteCostCalculator<T extends DefaultTripSchedule> implements CostCalculator<T> {
+
+  private final CostCalculator delegate;
+  private final Map<FeedScopedId, Integer> routePenalties;
+
+  public RouteCostCalculator(
+    @Nonnull CostCalculator<T> delegate,
+    @Nonnull Map<FeedScopedId, Integer> routePenalties
+  ) {
+    this.routePenalties = routePenalties;
+    this.delegate = delegate;
+  }
+
+  @Override
+  public int boardingCost(
+    boolean firstBoarding,
+    int prevArrivalTime,
+    int boardStop,
+    int boardTime,
+    T trip,
+    RaptorTransferConstraint transferConstraints
+  ) {
+    int defaultCost = delegate.boardingCost(
+      firstBoarding,
+      prevArrivalTime,
+      boardStop,
+      boardTime,
+      trip,
+      transferConstraints
+    );
+
+    int routeReluctanceCost = routePenalties.getOrDefault(trip.routeId(), ZERO_COST);
```

Comment:
Sorry, this is a total nit-pick but I want to avoid confusion: can you remove the word "reluctance" from the local variable name.

A reluctance is a cost that is proportional to the length of a leg. A penalty is applied only once.

Reply:
Good point! Actually, at some point, the idea was to apply unfererence per transit leg length and variable naming is leftover.

Reply:
To me the reluctance would actually be more useful, but I understand that you're just restoring the behaviour that was there before.

If I every have the need to unprefer routes I will implement the reluctance.